standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/RawStore.java

```diff
@@ -160,10 +160,11 @@
    * @param catName name of the catalog to alter.
    * @param cat new version of the catalog.
    * @throws MetaException something went wrong, usually in the database.
+   * @throws NoSuchObjectException no catalog of this name exists.
    * @throws InvalidOperationException attempt to change something about the catalog that is not
    * changeable, like the name.
    */
-  void alterCatalog(String catName, Catalog cat) throws MetaException, InvalidOperationException;
+  void alterCatalog(String catName, Catalog cat) throws MetaException, NoSuchObjectException, InvalidOperationException;
```

Comment:
This should also throw NoSuchObjectException if user is attempting to alter a non-existing catalog